docs/canvas/canvas-expression-lifecycle.asciidoc

```diff
@@ -0,0 +1,259 @@
+
+The expression does the following:
+* Retrieves the value from the *price* column in the first row of the `demodata` data table
+* Inputs the value to the `gte` function
+* Compares the value to `100`
+* Returns `true` if the value is 100 or greater, and `false` if the value is 100 or less
+
+That boolean value becomes the value for the `condition` argument. The output from the `then` expression is used as the output when `condition` is `true`. The output from the "else" expression is used when `condition` is false. In both cases, a base64-encoded image will be returned, and one of the two images will be displayed.
+
+You might be wondering how the <<getCell_fn, getCell>> function in the sub-expression accessed the data from the <<demodata_fn, demoData>> function, even though <<demodata_fn, demoData>> was not being directly piped into <<getCell_fn, getCell>>. The answer is simple, but important to understand. When nested sub-expressions are executed, they automatically receive the same _context_, or output of the previous function, that its parent function receives. In this specific expression, demodata’s data table is automatically provided to the nested expression’s `getCell` function, which allows that expression to pull out a value and compare it to another value. 
+
+The passing of the _context_ is automatic, and it happens no matter how deeply you nest your sub-expressions. To demonstrate this, let’s modify the expression slightly to compare the value of the price against multiple conditions using the <<all_fn, all>> function. 
+
+[source,text]
+----
+demodata
+| image dataurl={
+  if condition={getCell price | all {gte 100} {neq 105}} 
+    then={asset 3cb3ec3a-84d7-48fa-8709-274ad5cc9e0b} 
+    else={asset cbc11a1f-8f25-4163-94b4-2c3a060192e7}
+}
+----
+
+This time, `getCell price` is run, and the result is passed into the next function as the context.  Then, each sub-expression of the <<all_fn, all>> function is run, with the context given to their parent, which in this case is the result of `getCell price`. If `all` of these sub-expressions evaluate to `true`, then the `if` condition argument will be true. 
+
+Sub-expressions can seem a little foreign, especially if you aren’t a developer, but they’re worth getting familiar with, since they provide a ton of power and flexibility. Since you can nest any expression you want, you can also use this behavior to mix data from multiple indices, or even data from multiple sources. As an example, you could query an API for a value to use as part of the query provided to <<essql_fn, essql>>. 
+
+This whole section is really just scratching the surface, but hopefully after reading it you at least understand how to read expressions and make sense of what they are doing. With a little practice, you’ll get the hang of mixing _context_ and sub-expressions together to turn any input into your desired output.
```

Comment:
Add a comma between `it` and `you`, so it will look like:

This whole section is really just scratching the surface, but hopefully after reading it, you at least understand how to read expressions and make sense of what they are doing.